neo/SmartContract/InteropService.cs

```diff
@@ -200,13 +249,17 @@ private static bool Runtime_CheckWitness(ApplicationEngine engine)
 
         private static bool Runtime_Notify(ApplicationEngine engine)
         {
-            engine.SendNotification(engine.CurrentScriptHash, engine.CurrentContext.EvaluationStack.Pop());
+            StackItem state = engine.CurrentContext.EvaluationStack.Pop();
+            if (!CheckItemForNotification(state)) return false;
+            engine.SendNotification(engine.CurrentScriptHash, state);
             return true;
         }
 
         private static bool Runtime_Log(ApplicationEngine engine)
         {
-            string message = Encoding.UTF8.GetString(engine.CurrentContext.EvaluationStack.Pop().GetByteArray());
+            byte[] state = engine.CurrentContext.EvaluationStack.Pop().GetByteArray();
+            if (state.Length > MaxNotificationSize) return false;
```

Comment:
This `state` already was checked in the VM and never come back to the `EvaluationStack`. We need to check it again?

Reply:
This is an extra check for `Notify` and `Log`.